### DISCUSSION TITLE
SH-ESP32NMEA2000 gateway, does it pass unknown PGN's

### DISCUSSION THREAD

**Discussion by Asw1n** [hatlabs/discussions#49] [Q&A]
I am going to use my SH-ESP to have a gateway between my NMEA2000 and a signalK server using USB. The SignalK server generates a PGN (130824) that is not defined on N2KMessages.h. Will the gateway sofware still put this PGN on the NMEA2000 network. 

To be clear. The SH_ESP does not generate or process this message, it will just have to pass it on.

**Comment by mairas** [comment]
Hi, yes, as long as the messages are formatted as valid NMEA 2000 (J1939) messages, it shouldn't care about the contents.
